component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,321 @@
+                                "An event is created with the filepath, filename and " +
+                                "status of the file. Then that will be received by the FooStream. If there " +
+                                "are any changes a new event will be generated in every 200 milliseconds.\n"
+                ),
+        }
+)
+
+public class FileHandler extends Source<FileHandler.FileHandlerState> {
+    private static final Logger log = Logger.getLogger(FileHandler.class);
+    private static final String EMPTY_STRING = "";
+    private SourceEventListener sourceEventListener;
+    private long monitoringInterval = 100;
+    private String listeningDirUri;
+    private FileAlterationMonitor monitor;
+    private Map<String, Long> fileObjectMap = new ConcurrentHashMap<>();
+    private static final String CURRENT_MAP_KEY = "current.map.key";
+    private List<String> fileObjectList;
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public StateFactory<FileHandlerState> init(SourceEventListener sourceEventListener, OptionHolder optionHolder,
+                                               String[] requiredProperties, ConfigReader configReader,
+                                               SiddhiAppContext siddhiAppContext) throws SiddhiAppValidationException {
+        this.sourceEventListener = sourceEventListener;
+        if (optionHolder.isOptionExists(Constants.URI)) {
+            listeningDirUri = optionHolder.validateAndGetStaticValue(Constants.URI);
+        }
+        //Validation for URI
+        if (listeningDirUri == null || listeningDirUri.isEmpty()) {
+            throw new SiddhiAppCreationException("URI must be provided.");
+        }
+        FileObject listeningFileObject = Utils.getFileObject(listeningDirUri);
+        try {
+            if (!listeningFileObject.exists()) {
+                throw new SiddhiAppCreationException("Directory " + listeningFileObject.getPublicURIString()
+                        + " is not found.");
+            }
+            if (listeningFileObject.isFile()) {
+                throw new SiddhiAppCreationException("URI must belongs to a folder");
+            }
+            listeningDirUri = listeningFileObject.getName().getPath();
+        } catch (FileSystemException e) {
+            throw new SiddhiAppValidationException("Directory " + listeningFileObject.getPublicURIString()
+                    + " is not found.", e);
+        }
+
+        // Validation for fileNameList
+        String fileNameList = optionHolder.validateAndGetStaticValue(Constants.FILE_NAME_LIST, EMPTY_STRING);
+        fileNameList = fileNameList.replaceAll("\\s", "");
+        fileObjectList = Arrays.asList(fileNameList.split(","));
+        for (int i = 0; i < fileObjectList.size(); i++) {
+            String fileObjectPath = listeningDirUri + "/" + fileObjectList.get(i);
```

Comment:
@Ajanthy we may need to check the listeningDirUri ends with a "/" and if it is, we don't have to add "/" specifically.
If we add an extra "/", the 'if (fileObjectList.contains(file.getAbsolutePath())) {' in the Util. getFileHandlerEvent() method will fail.

Reply:
FileObject listeningFileObject = Utils.getFileObject(listeningDirUri);
listeningDirUri = listeningFileObject.getName().getPath();
Here the listeningDirUri is always without "/"